PICARD-1721: Add $join() script function and associated tests

rdswift opened the pull request:
<!--
    Hello! Thanks for submitting a pull request to MusicBrainz Picard. We
    appreciate your time and interest in helping our project!

    Use this template to help us review your change. Not everything is required,
    depending on your change. Keep or delete what is relevant for your change.
    Remember that it helps us review if you give more helpful info for us to
    understand your change.

    Ensure that you've read through and followed the Contributing Guidelines, in
    [CONTRIBUTING.md](https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md).
-->

# Summary

Add a scripting function to join the elements of a multi-value variable into a string using a supplied join phrase. An example might be `$set(artist_string,$join(%artists%,\, ))`

<!--
    Update the checkbox with an [x] for the type of contribution you are making.
-->

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**: Added a `join_multi()` scripting function to join the elements of a multi-value variable into a string using a supplied join phrase.  Also added associated tests.

# Problem
 Provide a way to quickly and easily join the elements of a multi-value variable into a string, using a custom join phrase between the elements.
<!--
    Anything that helps us understand why you are making this change goes here.
    What problem are you trying to fix? What does this change address?
-->

* JIRA ticket : PICARD-1721
<!--
    Please make sure you prefix your pull request title with 'PICARD-XXX' in order
    for our ticket tracker to link your pull request to the relevant ticket.
-->



# Solution

<!--
    The details of your change. Talk about technical details, considerations, or
    other interesting points. If you have a lot to say, be more detailed in this
    section.
-->
Added a `join()` scripting function to join the elements of a multi-value variable into a string using a supplied join phrase.  Also added associated tests.


# Action

<!--
    Other than merging your change, do you want / need us to do anything else
    with your change? This could include reviewing a specific part of your PR.
-->

zas commented on picard/script.py:
```diff
@@ -1066,6 +1066,28 @@ def func_map_multi(parser,
 
 
+def func_join_multi(parser,
+                       multi,
```
Indentation isn't correct here and below lines

rdswift replied:
I think I have it fixed now.  Sorry about that.

phw commented:
It probably makes sense to merge #1453 first with the decorator, then adapt this PR to it.

phw commented on picard/script.py:
```diff
@@ -1146,3 +1146,23 @@ def func_map(parser, multi, loop_code, separator=MULTI_VALUED_JOINER):
+
+
+@script_function(eval_args=False)
+def func_joinmulti(parser, multi, join_phrase, separator=MULTI_VALUED_JOINER):
```
I would be in favor of using `$join` as the function name here.

zas replied:
I'm in favor of `$join` too

zas commented:
@rdswift can you rebase and update PR subject/description ? (getting rid of $join_multi / $joinmulti occurences)

rdswift commented:
Updated and rebased.  Looks like one test is failing, but not due to the commit.  Can someone rerun the tests, or do I need to do something (close and reopen the PR or push another commit)?

zas commented:
> . Can someone rerun the tests

I did, one build failed due to a network issue, solved now.
Thanks for the rebase ;)
